### PR TITLE
avatar bug fixes and bring back user interaction

### PR DESCRIPTION
**Pull request by gtalha07**
- [x] fix parent badge not showing in avatars of `My Spaces` section.
- [x] Optimise parent badge size for UI.
- [x] bring back avatar and parent badge interaction.

### CODE REVIEW

File: app/lib/features/space/widgets/space_header_profile.dart [hunk 1]
```diff
@@ -42,8 +42,20 @@ class SpaceHeaderProfile extends ConsumerWidget {
...
+                    'spaceId': spaceId,
+                  },
+                ),
+                onParentBadgeTap: () => context.pushNamed(
+                  Routes.space.name,
+                  pathParameters: {
+                    'spaceId':
+                        canonicalParent.valueOrNull!.space.getRoomIdStr(),
+                  },
+                ),
```
Comment (kumarpalsinh25) [inline app/lib/features/space/widgets/space_header_profile.dart:57]: Are we using `onParentBadgeTap` in space header only? I mean I don't see that in other pages like RoomAvatar or widget where we are showing spaceBadge.

File: util/conv_test_man/ios/Flutter/Debug.xcconfig [hunk 2]
```diff
@@ -1 +1,2 @@
+#include? "Pods/Target Support Files/Pods-Runner/Pods-Runner.debug.xcconfig"
```
Comment (kumarpalsinh25) [inline util/conv_test_man/ios/Flutter/Debug.xcconfig:1]: Is this are also required changes relevant to this PR? If not then better to revert it. This applies to all the other configuration files that are changed.

File: app/macos/Podfile.lock [hunk 3]
```diff
@@ -100,4 +100,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: ce7dbe26c78bfc7ba46736094c1e2d25982870fa
 
-COCOAPODS: 1.13.0
\ No newline at end of file
+COCOAPODS: 1.14.2
```
[the remark this answers is not in the document]
Comment (gtalha07) [reply app/macos/Podfile.lock:103]: Yeah, its not required. Thanks for noticing!

File: util/conv_test_man/ios/Flutter/Debug.xcconfig (hunk 2 above)
Comment (gtalha07) [reply util/conv_test_man/ios/Flutter/Debug.xcconfig:1]: No, this'll be removed in revised commit :)

File: app/lib/features/space/widgets/space_header_profile.dart (hunk 1 above)
Comment (gtalha07) [reply app/lib/features/space/widgets/space_header_profile.dart:57]: Do we need to give interaction for parent badge everywhere? I need clarification on this. For chat list, it doesn't make sense and would be very weird and confusing if someone taps on room and takes to parent space.

**Pull request merged**